code/modules/wod13/onyxcombat.dm

```diff
@@ -578,3 +551,83 @@
 				elysium_checks = 0
 				if(HAS_TRAIT(src, TRAIT_ELYSIUM))
 					REMOVE_TRAIT(src, TRAIT_ELYSIUM, "elysium")
+
+/mob/living/carbon/human
+	var/current_police_raid = FALSE //lock to prevent multiple handle_police_raid() procs from going at the same time
+
+/mob/living/carbon/human/proc/handle_police_raid() //hooo boy
+	if(warrant)
+		if(key) //are we a player?
```

Comment:
we can use early returns to prevent all this nesting. 
```suggestion
		if(!key) //are we a player?
				last_nonraid = world.time
			    warrant = FALSE
			    return
```
Also, you might get more mileage checking for the presence of a human.mind datum for telling if a player is piloting things around.